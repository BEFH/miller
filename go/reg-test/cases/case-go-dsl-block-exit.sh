run_mlr --from $indir/s.dkvp put '$si = 0; for (i = 0; i < NR; i += 1) { if (i == 2) { $si += 0   } $si += i }'
run_mlr --from $indir/s.dkvp put '$si = 0; for (i = 0; i < NR; i += 1) { if (i == 2) { $si += 100 } $si += i }'
run_mlr --from $indir/s.dkvp put '$si = 0; for (i = 0; i < NR; i += 1) { if (i == 2) { break }      $si += i }'
run_mlr --from $indir/s.dkvp put '$si = 0; for (i = 0; i < NR; i += 1) { if (i == 2) { continue }   $si += i }'

run_mlr --from $indir/s.dkvp --opprint put '
  $si = 0;
  for (i = 0; i < NR; i += 1) {
    if (true) {
      if (i == 2) {
        $si += 0
      }
    }
    $si += i
  }'

run_mlr --from $indir/s.dkvp --opprint put '
  $si = 0;
  for (i = 0; i < NR; i += 1) {
    if (true) {
      if (i == 2) {
        $si += 100
      }
    }
    $si += i
  }'

run_mlr --from $indir/s.dkvp --opprint put '
  $si = 0;
  for (i = 0; i < NR; i += 1) {
    if (true) {
      if (i == 2) {
        break
      }
    }
    $si += i
  }'

run_mlr --from $indir/s.dkvp --opprint put '
  $si = 0;
  for (i = 0; i < NR; i += 1) {
    if (true) {
      if (i == 2) {
        continue
      }
    }
    $si += i
  }'

run_mlr --from $indir/s.dkvp --opprint put '
  $si = 0;
  for (p = 1; p <= 3; p += 1) {
    for (i = 0; i < NR; i += 1) {
      if (i == 2) {
        $si += 0
      }
      $si += i * 10**p
    }
  }'

run_mlr --from $indir/s.dkvp --opprint put '
  $si = 0;
  for (p = 1; p <= 3; p += 1) {
    for (i = 0; i < NR; i += 1) {
      if (i == 2) {
        break
      }
      $si += i * 10**p
    }
  }'

run_mlr --from $indir/s.dkvp --opprint put '
  $si = 0;
  for (p = 1; p <= 3; p += 1) {
    for (i = 0; i < NR; i += 1) {
      if (i == 2) {
        continue
      }
      $si += i * 10**p
    }
  }'
