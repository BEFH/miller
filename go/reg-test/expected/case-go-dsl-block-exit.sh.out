
mlr --from ./reg-test/input/s.dkvp put $si = 0; for (i = 0; i < NR; i += 1) { if (i == 2) { $si += 0   } $si += i }
a=pan,b=pan,i=1,x=0.3467901443380824,y=0.7268028627434533,si=0
a=eks,b=pan,i=2,x=0.7586799647899636,y=0.5221511083334797,si=1
a=wye,b=wye,i=3,x=0.20460330576630303,y=0.33831852551664776,si=3
a=eks,b=wye,i=4,x=0.38139939387114097,y=0.13418874328430463,si=6

mlr --from ./reg-test/input/s.dkvp put $si = 0; for (i = 0; i < NR; i += 1) { if (i == 2) { $si += 100 } $si += i }
a=pan,b=pan,i=1,x=0.3467901443380824,y=0.7268028627434533,si=0
a=eks,b=pan,i=2,x=0.7586799647899636,y=0.5221511083334797,si=1
a=wye,b=wye,i=3,x=0.20460330576630303,y=0.33831852551664776,si=103
a=eks,b=wye,i=4,x=0.38139939387114097,y=0.13418874328430463,si=106

mlr --from ./reg-test/input/s.dkvp put $si = 0; for (i = 0; i < NR; i += 1) { if (i == 2) { break }      $si += i }
a=pan,b=pan,i=1,x=0.3467901443380824,y=0.7268028627434533,si=0
a=eks,b=pan,i=2,x=0.7586799647899636,y=0.5221511083334797,si=1
a=wye,b=wye,i=3,x=0.20460330576630303,y=0.33831852551664776,si=1
a=eks,b=wye,i=4,x=0.38139939387114097,y=0.13418874328430463,si=1

mlr --from ./reg-test/input/s.dkvp put $si = 0; for (i = 0; i < NR; i += 1) { if (i == 2) { continue }   $si += i }
a=pan,b=pan,i=1,x=0.3467901443380824,y=0.7268028627434533,si=0
a=eks,b=pan,i=2,x=0.7586799647899636,y=0.5221511083334797,si=1
a=wye,b=wye,i=3,x=0.20460330576630303,y=0.33831852551664776,si=1
a=eks,b=wye,i=4,x=0.38139939387114097,y=0.13418874328430463,si=4

mlr --from ./reg-test/input/s.dkvp --opprint put 
  $si = 0;
  for (i = 0; i < NR; i += 1) {
    if (true) {
      if (i == 2) {
        $si += 0
      }
    }
    $si += i
  }
a   b   i x                   y                   si
pan pan 1 0.3467901443380824  0.7268028627434533  0
eks pan 2 0.7586799647899636  0.5221511083334797  1
wye wye 3 0.20460330576630303 0.33831852551664776 3
eks wye 4 0.38139939387114097 0.13418874328430463 6

mlr --from ./reg-test/input/s.dkvp --opprint put 
  $si = 0;
  for (i = 0; i < NR; i += 1) {
    if (true) {
      if (i == 2) {
        $si += 100
      }
    }
    $si += i
  }
a   b   i x                   y                   si
pan pan 1 0.3467901443380824  0.7268028627434533  0
eks pan 2 0.7586799647899636  0.5221511083334797  1
wye wye 3 0.20460330576630303 0.33831852551664776 103
eks wye 4 0.38139939387114097 0.13418874328430463 106

mlr --from ./reg-test/input/s.dkvp --opprint put 
  $si = 0;
  for (i = 0; i < NR; i += 1) {
    if (true) {
      if (i == 2) {
        break
      }
    }
    $si += i
  }
a   b   i x                   y                   si
pan pan 1 0.3467901443380824  0.7268028627434533  0
eks pan 2 0.7586799647899636  0.5221511083334797  1
wye wye 3 0.20460330576630303 0.33831852551664776 1
eks wye 4 0.38139939387114097 0.13418874328430463 1

mlr --from ./reg-test/input/s.dkvp --opprint put 
  $si = 0;
  for (i = 0; i < NR; i += 1) {
    if (true) {
      if (i == 2) {
        continue
      }
    }
    $si += i
  }
a   b   i x                   y                   si
pan pan 1 0.3467901443380824  0.7268028627434533  0
eks pan 2 0.7586799647899636  0.5221511083334797  1
wye wye 3 0.20460330576630303 0.33831852551664776 1
eks wye 4 0.38139939387114097 0.13418874328430463 4

mlr --from ./reg-test/input/s.dkvp --opprint put 
  $si = 0;
  for (p = 1; p <= 3; p += 1) {
    for (i = 0; i < NR; i += 1) {
      if (i == 2) {
        $si += 0
      }
      $si += i * 10**p
    }
  }
a   b   i x                   y                   si
pan pan 1 0.3467901443380824  0.7268028627434533  0
eks pan 2 0.7586799647899636  0.5221511083334797  1110
wye wye 3 0.20460330576630303 0.33831852551664776 3330
eks wye 4 0.38139939387114097 0.13418874328430463 6660

mlr --from ./reg-test/input/s.dkvp --opprint put 
  $si = 0;
  for (p = 1; p <= 3; p += 1) {
    for (i = 0; i < NR; i += 1) {
      if (i == 2) {
        break
      }
      $si += i * 10**p
    }
  }
a   b   i x                   y                   si
pan pan 1 0.3467901443380824  0.7268028627434533  0
eks pan 2 0.7586799647899636  0.5221511083334797  1110
wye wye 3 0.20460330576630303 0.33831852551664776 1110
eks wye 4 0.38139939387114097 0.13418874328430463 1110

mlr --from ./reg-test/input/s.dkvp --opprint put 
  $si = 0;
  for (p = 1; p <= 3; p += 1) {
    for (i = 0; i < NR; i += 1) {
      if (i == 2) {
        continue
      }
      $si += i * 10**p
    }
  }
a   b   i x                   y                   si
pan pan 1 0.3467901443380824  0.7268028627434533  0
eks pan 2 0.7586799647899636  0.5221511083334797  1110
wye wye 3 0.20460330576630303 0.33831852551664776 1110
eks wye 4 0.38139939387114097 0.13418874328430463 4440

